src/Models/RaspberryPiDevice.ts

```diff
@@ -71,11 +72,9 @@ export class RaspberryPiDevice extends ContainerDeviceBase {
   async upload(): Promise<boolean> {
     const isRemote = RemoteExtension.isRemote(this.extensionContext);
     if (!isRemote) {
-      const res = await askAndOpenInRemote(
+      await askAndOpenInRemote(
           OperationType.Upload, this.channel, this.telemetryContext);
-      if (!res) {
-        return false;
-      }
+      return false;

```

Comment:
same question here

Reply:
Same as above.

Reply:
If not remote, the operation failed. But will prompt to suggest user to open in remote.